packages/itwin/grouping-mapping-widget/src/widget/components/ExtractionMessageModal.tsx

```diff
@@ -0,0 +1,125 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { Button, DefaultCell, Icon, Modal, ModalButtonBar, ModalContent, Table, tableFilters, Text } from "@itwin/itwinui-react";
+import React, { useEffect, useMemo, useState } from "react";
+import type { CreateTypeFromInterface } from "../utils";
+import type { ExtractionMessageData } from "./hooks/useMappingsOperations";
+import "./ExtractionMessageModal.scss";
+import { SvgClock } from "@itwin/itwinui-icons-react";
+import type { CellProps, CellRendererProps, Column } from "react-table";
+import { StatusIcon } from "./StatusIcon";
+import { ExtractionLogCustomFilter } from "./ExtractionLogCustomFilter";
+
+export interface ExtractionMessageModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  extractionMessageData: ExtractionMessageData[];
+  timestamp: string;
+}
+
+export const ExtractionMessageModal = ({ isOpen, onClose, extractionMessageData, timestamp }: ExtractionMessageModalProps) => {
+  const [formattedTimestamp, setFormattedTimestamp] = useState<string>("");
+  useEffect(() => {
+    formatTimestamp;
+  });
+  const translatedLabels = useMemo(() => ({
+    filter: "Filter",
+    clear: "Clear",
+  }), []);
+  const columns = useMemo(
+    (): Column<CreateTypeFromInterface<ExtractionMessageData>>[] => [
+      {
+        id: "category",
+        Header: "Category",
+        accessor: "category",
+        fieldType: "text",
+        Cell: ({ value }: CellProps<ExtractionMessageData>) => String(value),
+        Filter: ExtractionLogCustomFilter,
```

Comment:
Where does `Filter` and `filter` come from? Naming looks suspicious.

Reply:
To what I understand from the iTwin UIDoc, `Filter` is the JSX element that contains the filter options and selection type (such as checkboxes or radio buttons) since in this case it's a customized filter. `filter` is the type of comparison, such as "text" for string matching and "equals" for equivalent values.